docs/src/main/paradox/scala/http/server-side/server-https-support.md

```diff
@@ -47,37 +55,63 @@ keystores using the JDK keytool utility can be found [here](https://docs.oracle.
+
+Java
+:  @@snip [SimpleServerApp.java](../../../../../../../akka-http-tests/src/main/java/akka/http/javadsl/server/examples/simple/SimpleServerApp.java) { #https-http-config }
+
+@scala[Once you configured the HTTPS context, you can set it as default:]
+@java[Then pass it to the `akka.http.javadsl.Http` class's `setDefaultServerHttpContext` method, like in the below `main` method.]
 
-@@snip [SimpleServerApp.java](../../../../../../akka-http-tests/src/main/java/akka/http/javadsl/server/examples/simple/SimpleServerApp.java) { #https-http-config }
+Scala
+:  @@snip [HttpsServerExampleSpec.scala](../../../../../test/scala/docs/http/scaladsl/server/HttpsServerExampleSpec.scala) { #set-low-level-context-default }
 
-Then pass it to `akka.http.javadsl.Http` class's `setDefaultServerHttpContext` method, like in the below `main` method.
+Java
+: @@snip [SimpleServerApp.java](../../../../../../../akka-http-tests/src/main/java/akka/http/javadsl/server/examples/simple/SimpleServerApp.java) { #https-http-app }
 
-@@snip [SimpleServerApp.java](../../../../../../akka-http-tests/src/main/java/akka/http/javadsl/server/examples/simple/SimpleServerApp.java) { #https-http-app }
+@@@ div { .group-scala }
+
+It is also possible to pass in the context to specific `bind...` (or client) calls, like displayed below:
+
+@@snip [HttpsServerExampleSpec.scala](../../../../../test/scala/docs/http/scaladsl/server/HttpsServerExampleSpec.scala) { #bind-low-level-context }
+
+@@@
 
 ## Running both HTTP and HTTPS
 
 If you want to run HTTP and HTTPS servers in a single application, you can call `bind...` methods twice,
 one for HTTPS, and the other for HTTP.
 
-When configuring HTTPS, you can do it up like explained in the above [Using HTTPS](#using-https-java) section,
+When configuring HTTPS, you can do it up like explained in the above [Using HTTPS](#using-https) section,
 
-@@snip [SimpleServerApp.java](../../../../../../akka-http-tests/src/main/java/akka/http/javadsl/server/examples/simple/SimpleServerApp.java) { #https-http-config }
+Scala
+:  @@snip [HttpsServerExampleSpec.scala](../../../../../test/scala/docs/http/scaladsl/server/HttpsServerExampleSpec.scala) { #low-level-default }
 
-or via [SSL-Config](#ssl-config-java) (not explained here though).
+Java
+:  @@snip [SimpleServerApp.java](../../../../../../../akka-http-tests/src/main/java/akka/http/javadsl/server/examples/simple/SimpleServerApp.java) { #https-http-config }
+
+or via [SSL-Config](#ssl-config) (not explained here though).
 
 Then, call `bind...` methods twice like below.
-The blow `SimpleServerApp.useHttps(system)` is calling the above defined HTTP `public static HttpsConnectionContext useHttps(ActorSystem system)` method.
+@scala[The passed `https` context is from the above code snippet.]
+@java[The blow `SimpleServerApp.useHttps(system)` is calling the above defined HTTP `public static HttpsConnectionContext useHttps(ActorSystem system)` method.]
```

Comment:
Oh, yes, I forgot to add a note about this one. "Below" sounds right, but we can reword to: 

> `SimpleServerApp.useHttps(system)` is calling the above defined ...